### PR TITLE
add `box-cas!` and `vector-cas!`

### PR DESCRIPTION
**Pull request by mflatt**
I think this change is mostly straightforward, but is `(set! ,(make-live-info) ,%rax (asm ,info ,asm-locked-cmpxchg ...))` on line 934 of "x86_64.ss" (and similar for "x86.ss") the best way to reflect that CMPXCHG may modify the RAX register?

### CODE REVIEW

**Comment by dybvig** [comment]
I don't think it's necessary since the preceding instruction kills `%rax` via `urax`, so I would remove the assignment and leave the RHS.

If it were necessary, I think you'd want it to set `urax` instead of `%rax`, since all assignments other than unspillable assignments need to be visible prior to live analysis.

**Comment by mflatt** [comment]
Updated.

**Comment by mflatt** [comment]
Rebased.

File: csug/objects.stex [hunk 1]
```diff
@@ -1,4 +1,4 @@
-% Copyright 2005-2017 Cisco Systems, Inc.
+<% Copyright 2005-2017 Cisco Systems, Inc.
```
Comment (burgerrg) [inline csug/objects.stex]: The initial < looks like a typo.

Comment (mflatt) [reply csug/objects.stex]: Fixed - thanks!

**Comment by burgerrg** [comment]
@mflatt, how do I compile this branch? I get `Error in $lookup-library-entry: entry ~s uninitialized
(150)`

**Comment by mflatt** [comment]
@burgerrg I use the usual(?) cross-compile steps:
```
set M=<platform>
export M
set ORIG=<current-version-build-dir>
export ORIG
(cd $M/s && make -f Mf-cross m=$M xm=$M base=$ORIG/$M)
make
```
But I always feel like I must be missing some better and simpler way.

**Comment by burgerrg** [comment]
Thanks, that worked! If you have a recent Chez Scheme installed, you can do this from the particular workarea:
```
(cd s && make all patchfile=patch Scheme=scheme SCHEMEHEAPDIRS=/usr/lib/csv%v/%m:/usr/local/lib/csv%v/%m)
make
```

**Comment by burgerrg** [comment]
Thanks for this pull request!

**Pull request merged**